src/settings/accounts-settings.tsx

```diff
@@ -37,194 +32,181 @@ import {
-
-	// Adds the email addresses of all the shared accounts
-	if (account.rights?.targets) {
-		account.rights?.targets.forEach((target) => {
-			if (target.target && (target.right === 'sendAs' || target.right === 'sendOnBehalfOf')) {
-				target.target.forEach((user) => {
-					if (user.type === 'account' && user.email) {
-						user.email.forEach((email) => {
-							result.push(email.addr);
-						});
-					}
-				});
+import { useAccountStore, useUserAccount, useUserSettings } from '../store/account';
+
+function mapToCreateIdentityRequests(
+	createRecord: Record<string, IdentityAttrs>
+): Array<CreateIdentityRequest> {
+	return map(
+		createRecord,
+		(item): CreateIdentityRequest => ({
+			_jsns: 'urn:zimbraAccount',
+			identity: {
+				name: item.zimbraPrefIdentityName,
+				_attrs: {
+					...item,
+					zimbraPrefFromAddressType: item.zimbraPrefFromAddressType || 'sendAs'
+				}
 			}
-		});
-	}
-
-	// Adds all the aliases
-	if (settings.attrs.zimbraMailAlias) {
-		if (isArray(settings.attrs.zimbraMailAlias)) {
-			result.push(...(settings.attrs.zimbraMailAlias as string[]));
-		} else {
-			result.push(String(settings.attrs.zimbraMailAlias));
-		}
-	}
-
-	return uniq(result);
-};
+		})
+	);
+}
 
-type IdentityAttrsRecord<T extends string | number> = Record<T, Partial<IdentityAttrs>>;
+function mapToDeleteIdentityRequests(deleteArray: Array<string>): Array<DeleteIdentityRequest> {
+	return map(
+		deleteArray,
+		(identityId, index): DeleteIdentityRequest => ({
+			_jsns: 'urn:zimbraAccount',
+			identity: { id: identityId },
+			requestId: index.toString()
+		})
+	);
+}
+
+function mapToModifyIdentityRequests(
+	modifyRecord: Record<string, Partial<IdentityAttrs>>
+): Array<ModifyIdentityRequest> {
+	return map(
+		modifyRecord.current,
+		(item, index): ModifyIdentityRequest => ({
+			_jsns: 'urn:zimbraAccount',
+			identity: {
+				id: index,
+				_attrs: item
+			}
+		})
+	);
+}
 
-export const AccountsSettings = ({
-	account,
-	identitiesDefault
-}: AccountSettingsProps): ReactElement => {
+export const AccountsSettings = (): JSX.Element => {
 	const [t] = useTranslation();
+	const createSnackbar = useSnackbar();
 
-	const [createList, setCreateList] = useState<NonNullable<IdentityAttrsRecord<number>>>({});
-	const [deleteList, setDeleteList] = useState<Array<string>>([]);
-	const [modifyList, setModifyList] = useState<NonNullable<IdentityAttrsRecord<string>>>({});
-
+	const account = useUserAccount();
 	const settings = useUserSettings();
 
-	const [delegatedSendSaveTarget, setDelegatedSendSaveTarget] = useState<
-		AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget']
-	>(settings.prefs.zimbraPrefDelegatedSendSaveTarget);
+	const createRecord = useRef<Record<string, IdentityAttrs>>({});
+	const deleteArray = useRef<Array<string>>([]);
+	const modifyRecord = useRef<Record<string, Partial<IdentityAttrs>>>({});
+
+	const delegatedSendSaveTarget = useRef<AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget']>(
```

Comment:
Should the ref variables have the Ref suffix in their name? Btw nice job the removal of the useless states :heart_eyes: